### ISSUE TITLE
Upgrade to Django 1.11

### ISSUE THREAD

**Issue by genghisken** [genghisken/ps1#67]
Originally reported by: **Ken Smith (Bitbucket: [genghisken](https://bitbucket.org/genghisken), GitHub: [genghisken](https://github.com/genghisken))**

----------------------------------------

Django 1.11 (released April 2017) will be supported for the next 3 years.  Time to upgrade.  Some things deprecated in 1.10 will break the current code, so this needs to be done carefully.

----------------------------------------
- Bitbucket: https://bitbucket.org/genghisken/ps1/issue/67

**Comment by genghisken** [comment]
Upgrade is now extremely urgent.  No further development of the web interface can take place until we do the upgrade!

The following changes must be done to ALL currently available Django sites - even the retrospective frozen ones (e.g. FGSS, MD) must be upgraded.

The changes should be implemented BEFORE the upgrade to Django 1.11

Initial testing on Mac indicates that these are the only changes required.

1. Addition of TEMPLATES in settings.py.  This replaces TEMPLATE_DIRS and TEMPLATE_CONTEXT_PROCESSORS.

E.g. in settings.py

```
#TEMPLATE_DIRS = (
#    # Put strings here, like "/home/html/django_templates" or "C:/www/django/templates".
#    # Always use forward slashes, even on Windows.
#    # Don't forget to use absolute paths, not relative paths.
#    "/Users/kws/Documents/code/gitdev/ps1/code/web/atlas/templates",
#)

#TEMPLATE_CONTEXT_PROCESSORS = (
#    "django.contrib.auth.context_processors.auth",
#    "django.contrib.messages.context_processors.messages",
#    "django.core.context_processors.debug",
#    "django.core.context_processors.i18n",
#    "django.core.context_processors.media",
#    "django.core.context_processors.request",
#    "django.core.context_processors.static",
#    "django.core.context_processors.request",
#)

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [
            # insert your TEMPLATE_DIRS here
            "/Users/kws/Documents/code/gitdev/ps1/code/web/atlas/templates",
        ],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                # Insert your TEMPLATE_CONTEXT_PROCESSORS here or use this
                # list if you haven't customized them:
                "django.contrib.auth.context_processors.auth",
                "django.contrib.messages.context_processors.messages",
                "django.template.context_processors.debug",
                "django.template.context_processors.i18n",
                "django.template.context_processors.media",
                "django.template.context_processors.request",
                "django.template.context_processors.static",
                "django.template.context_processors.request",
            ],
        },
    },
]
```

Note the change of context_processors from django.core to django.template.

2. All calls to render_to_response in views.py must be changed to render. The 'request' object must be first and the RequestContext object deleted.  Likewise the import of the csrf code comes from django.template not django.core anymore.

E.g.:

```
return render_to_response('login.html', {'form': form}, context_instance=RequestContext(request))
```

becomes

```
return render(request, 'login.html', {'form': form})
```

and

```
#from django.shortcuts import render_to_response, get_object_or_404
from django.shortcuts import render, get_object_or_404
```

and

```
#from django.core.context_processors import csrf
from django.template.context_processors import csrf
```

**Comment by genghisken** [comment]
This has been implemented.
